### PR TITLE
Parsing Travis logs to extract failling log data

### PR DESCRIPTION
**Pull request by dquimper**
This PR also includes #15 

The goal of this PR is to parse Travis logs to extract flip-flopping tests occurrence and update JIRA ticket with the information.

I started the PR with lots of refactoring and testing. I also prevent the "Upgrade Build Results" comment to be added to the PR when the report is empty.

Then I added the LogAnalyzer which detects if a build is from a PR and has failures. Then it adds a comment to the PR. The actual logs parsing and JIRA update is done in another project.

If you want to see what the PR comment looks like. There's an example here: https://github.com/Liaison-Intl/WebAdMIT/pull/7071

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,10 +1,12 @@
 source "https://rubygems.org"
 ruby "2.3.1"
 
-gem "builder"
+# gem "builder"
```
Comment (zefta) [inline Gemfile]: ??

Comment (dquimper) [reply Gemfile]: Hm... Don't remember doing that.

**Comment by dquimper** [comment]
@pfcarrier Can you help with getting this to pass Travis? There seems to have issues with `apk add`.

**Comment by pfcarrier** [comment]
hi @dquimper try to drop the constraint on the ca-certificate version.  That should do it.

https://github.com/Liaison-Intl/upgrade_toolbelt/blob/aca667b0effa708967545a5aa33f3052033b729a/Dockerfile.travis_analyzer#L9

eg remove the the "=2016...".  You can do the same to other module if any other still cause issue after that.  Pinning is good but sometime alpine drop package version due to security issue and this cause havoc like here.  Not a bad thing, this force us to upgrade.

**Review by cdarne — APPROVED** [review]

File: lib/travis_connection.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+class TravisConnection
+  attr_reader :github_token,
```
Comment (loto) [inline lib/travis_connection.rb]: Those getters/setters don't need to be public, you can either discard them and use the instance variable instead or make them private. I prefer to use the instance variable.

**Review by loto — APPROVED** [review]
You're free to ignore my comment, good job!

**Comment by dquimper** [comment]
@pfcarrier Can you confirm removing all the version pinning if ok?

https://github.com/Liaison-Intl/upgrade_toolbelt/pull/14/files#diff-88dbd98d360042611eaad412a03c5f10L4

**Comment by pfcarrier** [comment]
@dquimper :  ideally we would only remove the one we need to remove.  If you needed to remove all of them for it to work it's fine by me.  That said I feel not pinning ruby version is really bold.  Maybe we can pin the major.minor only to ensure we stay on 2.3?  Are we ok with ruby changing from 2.3 to 2.4 possibly without notice in the future ?

**Comment by dquimper** [comment]
Following @pfcarrier comment, creating a new PR to update the Dockerfile and rebased this branch.

**Review by kstevens715 — COMMENTED** [review]
This has been sitting in my list of review requests for the past 7 years. I think the only way to remove it is to approve this. I approve.

**Comment by pfcarrier** [comment]
wow, 7 years already!  Time flies @kstevens715 !

**Comment by kstevens715** [comment]
@pfcarrier Yeah, 7 years. Sorry it took so long to review. I hope no one was holding their breath 😆

**Pull request merged**